spdx/v3/v3_0/spdx_test.go

```diff
@@ -0,0 +1,214 @@
+package v3_0_test
+
+import (
+	"testing"
+
+	spdx "github.com/spdx/tools-golang/spdx/v3/v3_0"
+)
+
+func Test_exportImportExport(t *testing.T) {
+	// create a document
+	doc := spdx.NewDocument(
+		spdx.ProfileIdentifierType_Software,
+		"My Document",
+		&spdx.SoftwareAgent{Agent: spdx.Agent{Element: spdx.Element{
+			Name:    "tools-golang",
+			Summary: "a summary",
+		}}},
+		&spdx.Tool{Element: spdx.Element{
+			ExternalIdentifiers: spdx.ExternalIdentifierList{
+				&spdx.ExternalIdentifier{
+					ExternalIdentifierType: spdx.ExternalIdentifierType_Cpe23,
+					Identifier:             "cpe:2.3:a:myvendor:my-product:*:*:*:*:*:*:*:*",
+				},
+			},
+			ExternalRefs: nil,
+			Name:         "not-tools-golang",
+		}})
+
+	sbom := &spdx.SoftwareSbom{}
+	doc.RootElements.Append(sbom)
+
+	// create a package
+
+	pkg1 := &spdx.SoftwarePackage{
+		SoftwareSoftwareArtifact: spdx.SoftwareSoftwareArtifact{Artifact: spdx.Artifact{Element: spdx.Element{
+			Name: "some-package-1",
+		}}},
+		SoftwarePackageVersion: "1.2.3",
+	}
```

Comment:
I'm just going to call out the two least desirable traits of this proposal: moderately deep inheritance heirarchies with embedding are not especially fun to work with in initializers, since you must nest embedded types like this to set the "parent" type properties.